src/core/thread/tmf.hpp

```diff
@@ -57,8 +57,9 @@ class TmfAgent : public Coap::Coap
      * @param[in] aInstance      A reference to the OpenThread instance.
      *
      */
-    explicit TmfAgent(Instance &aInstance)
+    explicit TmfAgent(Instance &aInstance, otNetifIdentifier aNetifIdentifier)
```

Comment:
Is it possible to let this class inherit CoapBase and managing two sockets, one for Thread interface and one for Backbone link interface. I guess this will save some resources.

Reply:
It will save some RAM resource but the memory saved should be trivial for backbone routers which tends to have megabytes of RAM. 
With one COAP handling two sockets, we are going to make the COAP logic more complex with increased code size (e.g. choosing UDP sockets for sending/receiving messages based on message attributes)
Also with the security issue in mind, we will have to determine whether a resource should handle Backbone requests. 
I would prefer to use separate COAP objects since it's simpler. 
Thoughts?

Reply:
I think we can refactor in the future if `one COAP base two sockets` is proven to be a better idea.